### ISSUE TITLE
Unsigned integers as indices in DictEncoded type

### ISSUE THREAD

**Issue by dmbates** [JuliaData/Arrow.jl#113]
Per the discussion around https://julialang.slack.com/archives/C674VR0HH/p1611699007085600 the expectation is that unsigned integers in a `DataAPI.refarray` will be changed to signed integers when converting to `DictEncoded` but that doesn't seem to be happening.
```julia
julia> io = Arrow.write(IOBuffer(), (v = PooledArray(repeat(["a","b"], inner=5)),));

julia> tbl = Arrow.Table(seekstart(io))
Arrow.Table: (v = ["a", "a", "a", "a", "a", "b", "b", "b", "b", "b"],)

julia> typeof(tbl.v)
Arrow.DictEncoded{String, UInt8, Arrow.List{String, Int32, Vector{UInt8}}}
```

I notice that methods for `signedtype` are defined in src/arraytypes/dictencoding.jl but I don't seen any uses of that function.  It may be that this is vestigial and `encodingtype` is used instead.

**Comment by dmbates** [comment]
If I had to guess I would say that line 121 of src/arraytypes/dictencoding.jl, which is currently
```
            inds = copy(DataAPI.refarray(x))
```
should coerce to `Vector{signedtype(eltype(DataAPI.refarray(x))}` instead.

**Comment by quinnj** [comment]
Fixed in #119
